### PR TITLE
LoadAsync methods accepting string path.

### PR DESCRIPTION
**Pull request by pekspro**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Add LoadAsync methods that accepts `string path`.

This solves #1255

### CODE REVIEW

**Comment by pekspro** [comment]
I have added another unit test after this PR was created. Could the validation be run again? Or do I need to create a new PR?

**Comment by JimBobSquarePants** [comment]
Ah lovely, thanks for this! Will review ASAP. Don’t worry about additional commits, the CI system will automatically rebuild and test each time including everything as part of the same PR.

File: src/ImageSharp/Image.FromFile.cs [hunk 1]
```diff
@@ -114,6 +126,25 @@ public static Image Load(string path, out IImageFormat format)
...
+        /// Create a new instance of the <see cref="Image"/> class from the given file.
+        /// </summary>
+        /// <param name="configuration">The configuration for the decoder.</param>
+        /// <param name="path">The file path to the image.</param>
+        /// <exception cref="ArgumentNullException">The configuration is null.</exception>
+        /// <exception cref="ArgumentNullException">The path is null.</exception>
+        /// <exception cref="UnknownImageFormatException">Image format not recognised.</exception>
+        /// <exception cref="InvalidImageContentException">Image contains invalid content.</exception>
+        /// <returns>A <see cref="Task{Image}"/> representing the asynchronous operation.</returns>
+        public static async Task<Image> LoadAsync(Configuration configuration, string path)
```
Comment (antonfirsov) [inline src/ImageSharp/Image.FromFile.cs:139]: This one is untested.

Comment (pekspro) [reply src/ImageSharp/Image.FromFile.cs:139]: @antonfirsov, this is now added.

**Comment by TedStryker** [comment]
@pekspro
I changed my code to use your new LoadAsync overload tosurprisingly find, that your's is noticeably slower. 
When loading & resizing 2 random pictures (SaveAsync vs. SaveAsJpeg makes no difference), here is my "manual" benchmark (average after 30 executions):
Mine/Yours (ms):
300/550
180/300

I'm not so much into async I/O, so I have no explanation, but perhaps you have one and are able to further optimize this beautiful library.

My setup is Core 3.1 on Windows 8.1, my app is made with ASP.NET Core MVC.
The calls:
`
// My implementation:
return File(await image.LoadFromDisk(requestDimensions), "image/jpeg", $"{image.Name}-{requestDimensions}.jpg");
// Yours:
return File(await image.LoadFromDiskAsync(requestDimensions), "image/jpeg", $"{image.Name}-{requestDimensions}.jpg");
`

The code:
-Integration of your overload:
`            var outputStream = new MemoryStream();
            using (var image = await Image.LoadAsync(Configuration.Default, Url, Config.JpegDecoder)) {               
                    image.Mutate(i => i.Resize(new ResizeOptions {
                        Size = new Size(requestedDimensions[0], requestedDimensions[1]),
                        Mode = ResizeMode.BoxPad,
                        Sampler = Config.ImageResampler,
                    }));               

                await image.SaveAsync(outputStream, Config.ImageJpegEncoder);
                outputStream.Seek(0, SeekOrigin.Begin);`

-Mine:
`
            await using var fs = File.OpenRead(Url);
            var result = new byte[fs.Length];
            await fs.ReadAsync(result, 0, (int)fs.Length);

            using var image = Image.Load<Rgba32>(result, Config.JpegDecoder);
            image.Mutate(i => i.Resize(new ResizeOptions { /* same as above*/ }));

            var outputStream = new MemoryStream();
            image.SaveAsJpeg(outputStream, Config.ImageJpegEncoder);
            outputStream.Seek(0, SeekOrigin.Begin);`

**Comment by antonfirsov** [comment]
@TedStryker thanks for reporting this finding! It would be really helpful if you could open an issue for this performance problem. Conversations under closed issues/PR-s get lost and forgotten too often.

**Pull request merged**